CHANGELOG.md

```diff
@@ -18,6 +18,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 - Python: get the correct range when matching comprehension (#4221)
 - Python and other languages: allow matches of patterns containing
   non-ascii characters, but still with possibly many false positives (#4336)
+- Java: parse correctly constructor method patterns (#4418)
```

Comment:
If I understood it correctly, in general you can now write a method pattern without having to specify its type, which I see it's needed for matching constructors, but it can also be used for matching any other method.

Reply:
Yes, it could be abused for that.